POLIO-1762 POLIO-1764 POLIO-1765: Campaign combination cVDPV1 & cVDPV2 w/ 2 vaccines nOPV2 & bOPV

beygorghor opened the pull request:
We will have to create a new Virus Type: cVDPV1 & cVDPV2 with the Vaccine Type: nOPV2 & bOPV associated

Related JIRA tickets POLIO-1762

## Self proofreading checklist

- [ ] Did I use eslint and black formatters
- [ ] Is my code clear enough and well documented
- [ ] Are my typescript files well typed
- [ ] New translations have been added or updated if new strings have been introduced in the frontend
- [ ] My migrations file are included
- [ ] Are there enough tests
- [ ] Documentation has been included (for new feature)



## Changes

- added new constants
- implement new hashed colors as shape color

## How to test

- Open scope for a campaign, you should have nOPV2 & bOPV option in the vaccines.
- Create a campaign, try to select  cVDPV1 & cVDPV2 as virus.
- 

If a specific config is required explain it here: dataset, account, profile, etc.

## Print screen / video

![Screenshot 2024-12-17 at 15 26 13](https://github.com/user-attachments/assets/7a507dca-37fb-4b4e-aa78-d0be49b60e08)


## Follow the Conventional Commits specification

The **merge message** of a pull request must follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) specification.

This convention helps to automatically generate release notes.

Use lowercase for consistency.

[Example](https://github.com/BLSQ/iaso/commit/8b8d7d3064138c1e57878f17b4eb922516ab0112):

```
fix: empty instance pop up

Refs: IA-3665
```

Note that the Jira reference is preceded by a _line break_.

Both the line break and the Jira reference are entered in the _Add an optional extended description…_ field.

beygorghor committed:
adding constants

beygorghor committed:
Merge branch 'main' into POLIO-1762-nOPV2-bOPV-cVDPV2-cVDPV3

beygorghor committed:
adding virus type and vaccine type

beygorghor committed:
use color hashed shapes

beygorghor committed:
use hashed color

beygorghor committed:
lol

beygorghor committed:
glitch

quang-le committed:
Merge branch 'main' into POLIO-1762-nOPV2-bOPV-cVDPV2-cVDPV3

quang-le committed:
POLIO-1762: adapt vaccine max length in VRF

quang-le committed:
POLIO-1762: fix migrations

quang-le commented on hat/assets/js/apps/Iaso/components/maps/HashedPatterns/ColorsHashedPattern.tsx:
❤️

quang-le commented on plugins/polio/js/src/domains/VaccineModule/SupplyChain/constants.ts:
```diff
@@ -17,4 +17,8 @@ export const defaultVaccineOptions = [
         label: 'bOPV',
         value: 'bOPV',
     },
+    {
+        label: 'nOPV2 & bOPV',
+        value: 'nOPV2 & bOPV',
+    },
```
This is not correct: supplychain will still handle the vaccines separately. I'll open a separate ticket for that as it will require some backend as well I think

quang-le reviewed (changes requested):
The global feature still needs some spec and this PR will have to be released together with changes in the vaccine module, so I tagged is as DO NOT MERGE  in the meantime.

About the feature itself:
Nothing shocking in the code

- The hashed colors don't appear on the calendar

https://github.com/user-attachments/assets/1c89aadc-fcff-4322-9f4c-b7d00b492327

- They don't appear in the xlsx export of the calendar either:

![Screenshot 2024-12-18 at 12 17 18](https://github.com/user-attachments/assets/f1a8a085-96ca-455f-8ab8-c4a5a4c0db3a)

beygorghor committed:
Merge branch 'main' into POLIO-1762-nOPV2-bOPV-cVDPV2-cVDPV3

beygorghor committed:
add hashed color to calendar

beygorghor commented:
![Screenshot 2024-12-18 at 16 46 56](https://github.com/user-attachments/assets/6ad7c3ec-0347-4d3f-8a5b-fcb4a2eb2fbd)
@quang-le  i have adapted the calendar with the correct colors. Respected tickets have been created.

beygorghor committed:
hide columns

quang-le committed:
POLIO-1803: add model methods to return single vaccines

- On Round and Campaign
- Includes subactivities scopes

quang-le committed:
POLIO-1803: fix vaccine dropdown

- show individual vaccines for bOPV & nOPV2 campaigns
- Include subactivities vaccines

quang-le committed:
POLIO-1803: update vaccine filter dropdowns

quang-le committed:
POLIO-1803: fix minor bugs

quang-le committed:
fix: report type filter not working


Refs: POLIO-1805

beygorghor committed:
adapt reports and forms repository

quang-le committed:
POLIO-1803: add migration

quang-le committed:
fix: getDisplayNAme handles undefined user

beygorghor committed:
Merge pull request #1886 from BLSQ/POLIO-1803_adapt_suppplychain_to_bopv_nopv_new_vaccine

POLIO-1803: adapt supplychain to bopv/nopv new vaccine

quang-le committed:
POLIO-1762: remove empty lines in vaccine repo reports

quang-le committed:
fix typo
